src/Raven.Server/Documents/ETL/Providers/Queue/AmazonSqs/AmazonSqsEtl.cs

```diff
@@ -0,0 +1,268 @@
+        GetTransformer(DocumentsOperationContext context)
+    {
+        return new AmazonSqsDocumentTransformer<AmazonSqsItem>(Transformation, Database, context,
+            Configuration);
+    }
+
+    protected override int PublishMessages(List<QueueWithItems<AmazonSqsItem>> itemsPerQueue,
+        BlittableJsonEventBinaryFormatter formatter, out List<string> idsToDelete)
+    {
+        if (itemsPerQueue.Count == 0)
+        {
+            idsToDelete = null;
+            return 0;
+        }
+
+        var tooLargeDocsErrors = new Queue<EtlErrorInfo>();
+        idsToDelete = new List<string>();
+        int count = 0;
+
+        foreach (QueueWithItems<AmazonSqsItem> queue in itemsPerQueue)
+        {
+            string queueName = queue.Name.ToLower();
+
+            if (_queueClient == null)
+            {
+                _queueClient = QueueBrokerConnectionHelper.CreateAmazonSqsClient(
+                    Configuration.Connection.AmazonSqsConnectionSettings);
+            }
+
+            if (Configuration.SkipAutomaticQueueDeclaration == false &&
+                _alreadyCreatedQueues.ContainsKey(queueName) == false)
+            {
+                AsyncHelpers.RunSync(() => CreateQueue(_queueClient, queueName));
+            }
+
+            var batchMessages = new List<SendMessageBatchRequestEntry>();
+
+            foreach (AmazonSqsItem queueItem in queue.Items)
+            {
+                CancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    var sendMessageEntry = new SendMessageBatchRequestEntry
+                    {
+                        Id = queueItem.DocumentId,
+                        MessageBody = SerializeCloudEvent(queueItem)
+                    };
+
+                    batchMessages.Add(sendMessageEntry);
+
+                    if (batchMessages.Count == 10)
+                    {
+                        if (TrySendBatchMessages(queueName, batchMessages) == false)
+                        {
+                            // if batch sending failed, send each message individually
+                            SendMessagesOneByOne(queueName, batchMessages, queue, ref idsToDelete, ref tooLargeDocsErrors);
+                        }
+                        else
+                        {
+                            count += batchMessages.Count;
+                            if (queue.DeleteProcessedDocuments)
+                            {
+                                idsToDelete.AddRange(batchMessages.Select(entry => entry.Id));
+                            }
+                        }
+
+                        batchMessages.Clear();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new QueueLoadException($"Failed to prepare message, error reason: '{ex.Message}'", ex);
```

Comment:
_Suggestion_:
We're preparing, but also sending messages in this scope. Different verb may clarify the error messages expected here.
```suggestion
                    throw new QueueLoadException($"Failed to deliver message, error reason: '{ex.Message}'", ex);
```